jscomp/test/js_json_test.js

```diff
@@ -81,7 +81,7 @@ add_test("File \"js_json_test.res\", line 22, characters 11-18", (function (para
                   _0: false
                 };
         }
-        var ty2 = Js_json.classify(Caml_option.valFromOption(v$1));
+        var ty2 = Js_json.classify(v$1);
```

Comment:
This difference means the compiler now unboxes the optional.
Well OK it makes assumptions about values of json type being stored unboxed (in the sense of optionals).
While correct in this case, I think it's incidental. And the option unboxing mechanism needs to be taught about untagged variants.
I will investigate in a separate issue. No need to worry about that here.

Reply:
Good catch!